monai/transforms/intensity/dictionary.py

```diff
@@ -1020,6 +1017,58 @@ def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.nda
         return d
 
 
+class RandGibbsNoised(RandomizableTransform, MapTransform):
+    """
+    Dictionary-based wrapper for RandGibbsNoise.
+
+    Naturalistic image augmentation via Gibbs artifacts. The transform 
+    randomly applies Gibbs noise to 3D MRI images. Gibbs artifacts
+    are one of the common type of type artfifacts appearing in MRI scans.
+    
+    For general information on Gibbs artifacts, please refer to:
+    https://pubs.rsna.org/doi/full/10.1148/rg.313105115
+    https://pubs.rsna.org/doi/full/10.1148/radiographics.22.4.g02jl14949
+    
+    Args:
+        prob: probability of applying the transform.
+        keys: 'image', 'label', or ['image', 'label'] depending on which data
+                you need to transform.
+        alpha: Parametrizes the intensity of the Gibbs noise filter applied. Smaller
+            values of alpha correspond to milder applications. If a length-2 list is 
+            given as [a,b] then the value of alpha will be sampled uniformly from the 
+            interval [a,b]. 
+        allow_missing_keys: don't raise exception if key is missing.
+
+    """
+
+    def __init__(
+        self,
+        prob: float = 0.1,
+        keys: KeysCollection = "image",
+        alpha: Union[float, List[float]] = 0.5,
+        allow_missing_keys: bool = False
+    ) -> None:
+
+        assert prob <= 1 and prob >= 0, "prob must take values in [0,1]."
+        self.alpha = alpha
+
+        MapTransform.__init__(self, keys, allow_missing_keys)
+        RandomizableTransform.__init__(self, prob=prob)
+
+    def __call__(self, data: Mapping[Hashable, torch.tensor]) -> Dict[Hashable, torch.tensor]:
+
+        d = dict(data)
+        self.randomize(None)
+
+        if not self._do_transform:
+            return d
+        else:
+            for key in self.key_iterator(d):
+                gibbs = RandGibbsNoise(self.alpha, 1.0)
```

Comment:
I'm not sure this logic will work. You'll be creating a different random instance each time and so the image and label will have two different transforms applied to them. I think you'll want to have something in the constructor that creates a single transform: `self.transform = RandGibbsNoise(self.alpha, 1.0)`. And then in `randomize` you'll want to call `self.transform.randomize`.

Reply:
You could also simplify the if else logic to `if self._do_transform:  for key in ...` and then outdented you can have `return d` to reduce duplication (similar to what I suggested for the array transform).

Reply:
You are correct. If the user gives `alpha=[a,b]` then the two instances of `RandGibbsNoise` will have different `alpha` values.

Thanks for that catch!

Reply:
Yeah, sounds like separate deterministic and random versions may be necessary, sorry about that!